src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/BlockedInternalsBlockingPolicy.cs

```diff
@@ -278,6 +280,18 @@ public override bool IsBlocked(FieldDesc field)
                 {
                     return false;
                 }
+
+                // Exempt fields on custom attributes from blocking. We use a heuristic instead of
+                // unconditionally walking the base class hierarchy as a perf optimization.
```

Comment:
Is it really faster on averate to compare the names than just do `CanCastTo` ?

Reply:
Yeah, it might be premature. Just worried about a scenario where a generic type has a lot of fields, each of which is a deeply nested type.

This will currently only trigger in internals-reflection-blocked assemblies, which somewhat limits the fallout to framework only.